File: 2043. Simple Bank System.cpp
class Bank {
public:
    vector<long long> bal;
    Bank(vector<long long>& balance) {
        bal = balance;
    }
    bool isValidAccount(int account) {
        return account and account <= bal.size();
    }
    bool availableMoney(int &account, long money) {
        return bal[account - 1] >= money;
    }
    
    bool transfer(int account1, int account2, long long money) {
        if(not isValidAccount(account1) or not isValidAccount(account2))
            return false;
        
        if(not availableMoney(account1, money)) 
            return false;
        
        bal[account1 - 1] -= money;
        bal[account2 - 1] += money;
        return true;
    }

    
    bool deposit(int account, long long money) {
        if(not isValidAccount(account))
            return false;
        bal[account - 1] += money;
        return true;
    }
    
    bool withdraw(int account, long long money) {
        if(not isValidAccount(account))
            return false;
        if(not availableMoney(account, money)) 
            return false;

        bal[account - 1] -= money;
        return true;
    }
};

File: 2043. Simple Bank System.py
class Bank:

    def __init__(self, balance: List[int]):
        self.n = len(balance)
        self.balance = [-1] + balance

    def transfer(self, account1: int, account2: int, money: int) -> bool:
        if account1 > self.n or account2 > self.n:
            return False
        
        if self.balance[account1] < money:
            return False
        
        self.balance[account1] -= money
        self.balance[account2] += money
        return True

    def deposit(self, account: int, money: int) -> bool:
        if account > self.n:
            return False
        self.balance[account] += money
        return True

    def withdraw(self, account: int, money: int) -> bool:
        if account > self.n or self.balance[account] < money:
            return False
        self.balance[account] -= money
        return True